src/app/components/find-journey/find-journey.component.ts

```diff
@@ -23,17 +25,42 @@ export class FindJourneyComponent {
   }
 
   onSubmit() {
-    console.log('id searched: ', this.id.value);
-    console.log('destination searched: ', this.destination.value);
 
+    if (this.id.value !== undefined && this.id.value.trim().length !== 0) {
+      console.log('id searched: ', this.id.value);
+      this.searchById();
+    } else if (this.destination.value !== undefined && this.destination.value.trim().length !== 0) {
+      console.log('destination searched: ', this.destination.value);
+      this.searchByCriterias();
+    }
+  }
+
+  private searchByCriterias() {
+    this.journeyService.searchByCriterias(this.destination.value)
+      .pipe(
+        tap(result => log(result)),
+        filter(res => Array.from(res).length > 0),
+        tap(result => log(result))
+      )
+      .subscribe(
+        (response) => {
+          this.journeysFound = response;
+        },
+        (err) => console.error(err),
+        () => console.log('done searching by criteria !')
+      );
+  }
+
+  private searchById() {
     this.journeyService.searchById(this.id.value)
+      .pipe(tap(result => log(result)))
       .subscribe(
         (response) => {
           this.journey = response;
```

Comment:
I don't understand why we have journey and journeyFound?

Reply:
jounrey found is the list of journeys that were found whatever the search method is used.
jouneyr is the one found using the method searchById only.
We put the jouney found in the list so that we can display it.